docs/_extensions/doxyrunner.py

```diff
@@ -0,0 +1,399 @@
+"""
```

Comment:
This is 3rd party code taken from Zephyr (Apache-licensed, see below). It lets us run Doxygen as a step in the Sphinx build.